File: 26. Remove Duplicates from Sorted Array/remove_duplicates_from_sorted_array.py
def duplicates_shifting_remove (array):
	counter = 0
	len_array = len(array)
	
	array.sort()
	
	for i in range ((len_array - 1), -1, -1):
		element = array[i]
		unique = True
		
		for j in range (i):
			if element == array[j]:
				unique = False
				break
		
		if (unique == False):
			counter += 1
			holder = array[i]
			
			for j in range (i, (len_array - 1)):
				array[j] = array[j + 1]
				
			array[len_array - 1] = holder
			
	#array_a = array[:-counter].sort()
	#array[:-counter].sort()
	#array_b = array[-counter:].sort()
	#array[-counter:].sort()
	#array = array_a + array_b
	
	return counter

nums = [0,0,1,1,1,2,2,3,3,4]
val = duplicates_shifting_remove(nums)
print (val)
print ("\n\n\n")
print (nums)
print (nums[:-val])
print (nums[-val:])

File: 26. Remove Duplicates from Sorted Array/remove_duplicates_from_sorted_array.cpp
#include<iostream>
#include<algorithm>

using namespace std;

int duplicates_shifting_remove (int * list, int len_list){
	int counter = 0;
	
	sort (list, list + len_list);
	
	for (int i = (len_list - 1); i >= 0; i--){
		int element = list[i];
		bool unique = true;
		
		for (int j = 0; j < i; j++){
			if (element == list[j]){
				unique = false;
				break;
			}
		}
		
		if (unique == false){
			counter ++;
			int holder = list[i];
			
			for (int j = i; j < (len_list - 1); j++){
				list[j] = list[j + 1];
			}
			
			list[len_list - 1] = holder;
		}
	}
	
	return (counter);
}

int main () {
	int nums [] = {0,0,1,1,1,2,2,3,3,4};
	int len_nums = sizeof(nums)/ sizeof(nums[0]);
	
	int rem = duplicates_shifting_remove(nums, len_nums);
	
	cout << "This is the handy work of C++\n\n";
	cout << "The number of the duplicates in the array are:\t" << rem << "\n\n\n";

	printf("[");
	for (int i = 0; i < len_nums; i++){
		cout << nums[i];
		
		if (i == (len_nums - 1)){
			cout << "]" << endl;
		} else {
			cout << ", ";
		}
	}
	
	printf("\n\n\n[");
	for (int i = 0; i < (len_nums - rem); i++){
		cout << nums[i];
		
		if (i == ((len_nums - rem) - 1)){
			cout << "]\n";
		} else {
			cout << ", ";
		}
	}

	return (0);
}
